howso/train_ts_ablation.amlg

```diff
@@ -0,0 +1,562 @@
+			time_feature_index (get feature_index_map !tsTimeFeature)
+			id_features (get !tsModelFeaturesMap "series_id_features")
+			id_indices (unzip feature_index_map (get !tsModelFeaturesMap "series_id_features"))
+			original_features features
+		))
+		(declare (assoc
+			series_id_features (get !featureAttributes [(first lag_features) "auto_derive_on_train" "series_id_features"])
+			series_ordered_by_features (get !featureAttributes [(first lag_features) "auto_derive_on_train" "ordered_by_features"])
+			needs_grouping
+				(call !DoesDataNeedGroupingById (assoc
+					data cases
+					id_indices id_indices
+				))
+		))
+
+		(if needs_grouping
+			(assign (assoc cases (call !GroupDataByIds) ))
+		)
+
+		;the non-lag 'custom' derivation type features below
+		(assign (assoc
+			derived_features
+				(filter
+					(lambda
+						(and
+							(not (contains_value lag_features (current_value)))
+							(= "custom" (get !featureAttributes [(current_value 1) "auto_derive_on_train" "derive_type"]) )
+						)
+					)
+					derived_features
+				)
+		))
+
+		(if encode_features_on_train
+			(assign (assoc
+				cases
+					(map
+						(lambda
+							(call !ConvertFromInput (assoc
+								feature_values (current_value 1)
+								features features
+							))
+						)
+						cases
+					)
+			))
+		)
+
+		(declare (assoc
+			previous_ids []
+			start_index 0
+			end_index 0
+			num_rows (size cases)
+			output_case_ids []
+			;map of sorted case index -> original case index used to lookup which original rows were ablated after data is sorted
+			ts_ablated_indices_map (null)
+			;number of cases to be trained from each series, accumulated to keep track of session training index
+			num_previously_trained_cases 0
+		))
+
+		(while (< (current_index) num_rows)
+			(if (= 0 (current_index))
+				(assign (assoc previous_ids (unzip (get cases (current_index 1)) id_indices) ))
+			)
+
+			;encountering next id or at end of all cases, train on the previous list of cases
+			(if
+				(or
+					(!= previous_ids (unzip (get cases (current_index)) id_indices) )
+					(= (current_index) (- num_rows 1))
+				)
+				(let
+					(assoc
+						end_index
+							;if last index, should be as-is
+							(if (= (current_index 1) (- num_rows 1))
+								(current_index 1)
+								(- (current_index 1) 1)
+							)
+						features original_features
+					)
+
+					;train one entire series at a time
+					(accum (assoc
+						output_case_ids
+							(call !TrainTimeSeriesWithAblation (assoc
+								data (unzip cases (range start_index end_index))
+							))
+					))
+
+					(assign (assoc
+						previous_ids (unzip (get cases (current_index 1)) id_indices)
+						start_index (current_index 1)
+						num_previously_trained_cases (+ num_previously_trained_cases 1 (- end_index start_index))
+					))
+				)
+			)
+		)
+
+		;return trained case_ids
+		output_case_ids
+	)
+
+	;derive then train with ablation
+	#!TrainTimeSeriesWithAblation
```

Comment:
This method name is like the exact same as the other... If I am understaning the code correctly, I suggest we rename this one to something like #!TrainSingleSeriesWithAblation